camel/storages/vectordb_storages/milvusDB_v2.py

```diff
@@ -0,0 +1,99 @@
+    BaseVectorStorage,
+    VectorDBQuery,
+    VectorDBQueryResult,
+    VectorDBStatus,
+    VectorRecord,
+)
+
+class MilvusStorage(BaseVectorStorage):
+    """
+    This module is to integration of Milvus, a vector database, into Camel agent.
+    The information of Milvus can be accessed to Milvus websites at: https://milvus.io/docs
+
+    Args:
+        vector_dim (int): The dimenstion of storing vectors.
+        collection_name (Optional[str], optional): Name for the collection in
+            the Milvus. If not provided, set it to the current time with iso
+            format. (default: :obj:`None`)
+        url_and_api_key (Optional[Tuple[str, str]], optional): Tuple containing
+            the URL and API key for connecting to a remote Milvus instance.
+            (default: :obj:`None`)
+        path (Optional[str], optional): Path to a directory for initializing a
+            local Milvus client. (default: :obj:`None`)
+        distance (VectorDistance, optional): The distance metric for vector
+            comparison (default: :obj:`VectorDistance.COSINE`)
+        **kwargs (Any): Additional keyword arguments for initializing
+            `MilvusClient`.
+    """
+
+    # Initialize the MilvusStorage class with required parameters
+    def __init__(self, vector_dim: int, url_and_api: Tuple[str, str], **kwargs: Any) -> None:
+        self._client: MilvusClient
+        self.vector_dim = vector_dim
+        self.create_client(url_and_api, **kwargs)   
+        self.collection_name = self.create_collection()
+
+    # Create a Milvus client using the provided URL and API key
+    def create_client(self, url_and_api_key: Tuple[str, str], **kwargs: Any) -> None:
+        self._client = MilvusClient(uri=url_and_api_key[0], token=url_and_api_key[1], **kwargs)
+
+    # Create a new collection in Milvus with a unique name and vector dimensions
+    def create_collection(self, **kwargs: Any,) -> str:
+        try:    
+            collection_name  = self.generate_collection_name()
+            self._client.create_collection(collection_name = collection_name, dimension = self.vector_dim, **kwargs,)
+            return collection_name
+        except Exception as e:
+            print(f"Failed to create collection: {e}")
+            raise 
+
+    # Delete a specified collection from Milvus 
+    def delete_collection(self, collection_name: str) -> None:
+        self._client.drop_collection(collection_name=collection_name)
+
+    # Generate a unique collection name using UUID
+    def generate_collection_name(self) -> str:
+        return str(uuid.uuid4())
+
+    # Add vector records to the collection    
+    def add(self, records: List[VectorRecord], **kwargs) -> None:
+        try:
+            vectors = [record.vector for record in records]
+            ids = [record.id for record in records] if hasattr(records[0], 'id') else None
+            self._client.insert(collection_name=self.collection_name, entities=vectors, ids=ids, **kwargs)
```

Comment:
how did this work? `.insert` doesn't accpet entities and ids as input parameter

Reply:
In my latest modified script, I remodified this function and redefined the parameters，Thanks